src/ImageData/PolarizationCalculator.cc

```diff
@@ -0,0 +1,266 @@
+    blc(dir_axes[0]) = x_range.from;
+    trc(dir_axes[0]) = x_range.to;
+    blc(dir_axes[1]) = y_range.from;
+    trc(dir_axes[1]) = y_range.to;
+    blc(spectral_axis) = z_range.from;
+    trc(spectral_axis) = z_range.to;
+
+    // Get stokes indices and make stokes regions
+    if (coord_sys.hasPolarizationCoordinate()) {
+        const auto& stokes = coord_sys.stokesCoordinate();
+        int stokes_index;
+        if (stokes.toPixel(stokes_index, casacore::Stokes::I)) {
+            _stokes_images[I] = MakeSubImage(blc, trc, stokes_axis, stokes_index);
+        }
+        if (stokes.toPixel(stokes_index, casacore::Stokes::Q)) {
+            _stokes_images[Q] = MakeSubImage(blc, trc, stokes_axis, stokes_index);
+        }
+        if (stokes.toPixel(stokes_index, casacore::Stokes::U)) {
+            _stokes_images[U] = MakeSubImage(blc, trc, stokes_axis, stokes_index);
+        }
+        if (stokes.toPixel(stokes_index, casacore::Stokes::V)) {
+            _stokes_images[V] = MakeSubImage(blc, trc, stokes_axis, stokes_index);
+        }
+    } else { // Assume stokes indices: I = 0, Q = 1, U = 2, and V = 3
+        auto stokes_axis_size = _image->shape()[stokes_axis];
+        if (stokes_axis_size > 0) {
+            _stokes_images[I] = MakeSubImage(blc, trc, stokes_axis, 0);
+        }
+        if (stokes_axis_size > 1) {
+            _stokes_images[Q] = MakeSubImage(blc, trc, stokes_axis, 1);
+        }
+        if (stokes_axis_size > 2) {
+            _stokes_images[U] = MakeSubImage(blc, trc, stokes_axis, 2);
+        }
+        if (stokes_axis_size > 3) {
+            _stokes_images[V] = MakeSubImage(blc, trc, stokes_axis, 3);
+        }
+    }
+}
+
+void PolarizationCalculator::FiddleStokesCoordinate(casacore::ImageInterface<float>& image, casacore::Stokes::StokesTypes type) {
+    casacore::CoordinateSystem coord_sys = image.coordinates();
+    int stokes_index = coord_sys.findCoordinate(casacore::Coordinate::STOKES);
+    if (stokes_index > -1) {
+        casacore::Vector<int> which(1);
+        which(0) = (int)type;
+        casacore::StokesCoordinate stokes(which);
+        coord_sys.replaceCoordinate(stokes, stokes_index);
+        image.setCoordinateInfo(coord_sys);
+    }
+}
+
+std::shared_ptr<casacore::ImageInterface<float>> PolarizationCalculator::GetStokesImage(const StokesTypes& type) {
+    return _stokes_images[type];
+}
+
+casacore::LatticeExprNode PolarizationCalculator::MakeTotalPolarizedIntensityNode() {
+    casacore::LatticeExprNode lin_node = casacore::LatticeExprNode(
+        casacore::pow(*_stokes_images[V], 2) + casacore::pow(*_stokes_images[U], 2) + casacore::pow(*_stokes_images[Q], 2));
+    return casacore::sqrt(lin_node);
+}
+
+casacore::LatticeExprNode PolarizationCalculator::MakePolarizedIntensityNode() {
+    casacore::LatticeExprNode lin_node =
+        casacore::LatticeExprNode(casacore::pow(*_stokes_images[U], 2) + casacore::pow(*_stokes_images[Q], 2));
+    return casacore::sqrt(lin_node);
```

Comment:
I'd prefer to use `std::pow`, `std::sqrt`. Is there any reason not to use STL?

Reply:
It seems `std::pow` and `std::sqrt` can not process the variable type `casacore::LatticeExprNode`. So I did not change this part of the code.